### PR TITLE
[6.3 Feature] Canned Role Org Admin feature UI Stubs

### PR DESCRIPTION
**Pull request by jyejare**
$subject

### CODE REVIEW

File: tests/foreman/ui/test_role.py [hunk 1]
```diff
@@ -567,30 +567,42 @@ def test_positive_create_overridable_filter(self):
                 menu_locators['menu.hosts'], timeout=3))
 
     @stubbed
-    @tier2
-    def test_positive_update_role_taxonomies(self):
-        """Update role taxonomies which applies only to non-overrided filters
-        in role
+    @tier1
+    def test_negative_update_role_taxonomies_overridden_filters(self):
+        """Overridden role filters taxonomies doesnt updates on update of role
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: `Overridden role filters taxonomies doesnt updates on update of role taxonomies` is not readable enough from language perspective

File: tests/foreman/ui/test_role.py [hunk 2]
```diff
@@ -567,30 +567,42 @@ def test_positive_create_overridable_filter(self):
...
 
-            1. The role is updated successfully
-            2. User should have access to non-overrided resources of role in
-                updated taxonomies
-            3. User shouldnt have access to overrided resources of role in
-                updated taxonomies
+    @stubbed
+    @tier2
+    def test_positive_override_checkmark(self):
+        """Filter table displays checkmark image in override column for
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: Same here. I can understand steps and expected results, but not description

File: tests/foreman/ui/test_role.py [hunk 3]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
-            2. Assign taxonomies (organizations and locations) to the cloned
-                role
-            3. Add more missing resource permission to the cloned role to make
-                it Org Admin having access to all resources
-            4. Create user and assign org admin role
-            5. Login with user and attempt to access resources
-
-        :expectedresults:
+            1. create Org Admin by cloning 'Organization admin' role which has
+                most resources permissions
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: four spaces are needed

File: tests/foreman/ui/test_role.py [hunk 4]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
 
     @stubbed
-    @tier2
-    def test_positive_create_org_admin(self):
-        """Create Org admin role which has access to all the resources within
-        organization
+    @tier1
+    def test_positive_create_org_admin_from_clone(self):
+        """Create Org admin role which has access to most of the resources
+            within organization
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: don't think that spaces are needed here

File: tests/foreman/ui/test_role.py [hunk 5]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
     def test_positive_create_cloned_role_with_taxonomies(self):
-        """Test taxonomies can be set on cloned role.
+        """Taxonomies can be assigned to cloned role
 
         :id: ad20f5c7-3df7-4b43-8a52-097c87676d07
 
         :steps:
 
-            1. Attempt to clone any existing role(e.g Manager Role)
+            1. Create Org Admin by cloning 'Organization admin' role
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: `Org Admin` is a name or what?

File: tests/foreman/ui/test_role.py [hunk 6]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
 
         :CaseImportance: Critical
         """
 
     @stubbed
-    @tier1
+    @tier3
+    def test_positive_access_entities_from_org_admin(self):
+        """User can access resources within its taxonomies if assigned role
+            has permission for same taxonomies
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: no spaces are needed here

File: tests/foreman/ui/test_role.py [hunk 7]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
+
+        :id: dcdc438a-a8f0-4b3e-a8cb-e6f5c9215b1b
+
+        :steps:
+
+            1. Create org_admin and assign taxonomies to it
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: What is `org_admin` here?

File: tests/foreman/ui/test_role.py [hunk 8]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
+            2. Create user with same taxonomies as above org admin taxonomies
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: `Create user with same taxonomies as role above` sounds better.

File: tests/foreman/ui/test_role.py [hunk 9]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
+            3. Assign above cloned role to user.
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: `Assign cloned role to user above` sounds better.

File: tests/foreman/ui/test_role.py [hunk 10]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
+
+        :expectedresults: User should be able to access all the resources and
+            permissions in taxonomies selected in org admin role
+        """
+
+    @stubbed
+    @tier3
+    def test_negative_access_entities_from_org_admin(self):
+        """User can not access resources in taxonomies assigned to role if
+            its own taxonomies are not same as its role
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: no spaces needed

File: tests/foreman/ui/test_role.py [hunk 11]
```diff
@@ -660,113 +654,156 @@ def test_positive_disable_all_filters_overriding_option(self):
...
+            2. Overridden filter should have taxonomies assigned
+            3. Clone above role
+            4. View cloned role filters
 
         :expectedresults:
 
-            1. On cloning, taxonomies of the overridden filters in cloned role
-                are set to None
+            1. Taxonomies of the 'parent role overridden filters' are set to
+                None in cloned role
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: probably, 4 spaces neeeded

File: tests/foreman/ui/test_role.py [hunk 12]
```diff
@@ -775,23 +812,493 @@ def test_positive_empty_filter_taxonomies_in_cloned_role(self):
...
+
+            1. Both the user should list role inherited from usergroup
+            2. Both the user shouldn't have access to the resources of
+                organization A,B and Location A,B
+        """
+
+    @stubbed
+    @tier2
+    def test_negative_assign_taxonomies_by_org_admin(self):
+        """Org Admin doesnt have permissions to assign org/loc to any of
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: `Org Admin doesn't have permissions to assign org/loc to any of its entities`

File: tests/foreman/ui/test_role.py [hunk 13]
```diff
@@ -775,23 +812,493 @@ def test_positive_empty_filter_taxonomies_in_cloned_role(self):
...
+            4. Delete the org admin role
+
+        :expectedresults: Super Admin should be able to remove org admin role
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_taxonomies_control_to_superAdmin_with_org_admin(self):
+        """Super Admin can access entities in taxonomies assigned to Org Admin
+        in org admin's presence
```
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: can you please clarify word `presence` here?

File: tests/foreman/ui/test_role.py (hunk 3 above)
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: Didnt get that... Where do we need 4 spaces ?

File: tests/foreman/ui/test_role.py (hunk 5 above)
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: The org admin isnt a role, but we will create a role that will have all the permissions for perticular org(s) and location(s).
In short, Org Admin is shortname for Admin that controls the Org.

Comment (jyejare) [inline tests/foreman/ui/test_role.py]: You can create org admin by cloning 'Organization admin' role, and assign org and location to that cloned role on which the user will have control.

Comment (jyejare) [inline tests/foreman/ui/test_role.py]: To create an Org Admin both the steps in this test is required.

File: tests/foreman/ui/test_role.py (hunk 7 above)
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: Same as above explained org admin, I am renaming it to have the same nomenclature.

File: tests/foreman/ui/test_role.py (hunk 11 above)
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: I didnt understand this but make test-docstrings doesnt throw any exception for this

File: tests/foreman/ui/test_role.py (hunk 13 above)
Comment (jyejare) [inline tests/foreman/ui/test_role.py]: Presence in the sence, Super Admin can access the entities in Org Admin taxonomies even though the Org Admin present(exists/not_deleted). 
In short what I want to test is - > The control of taxonomies are not completely taken by Org Admin from Super Admin, Super Admin can still access what Org Admin can access
Simply removing the presence

**Comment by jyejare** [comment]
Hey @oshtaier I fixed all the necessary comments and replied on your questions.
Also I have added 5 new testcases for LDAP authantication in bottom of the file.

Please revisit again, Thanks.

File: tests/foreman/ui/test_role.py (hunk 3 above)
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: sorry for that

File: tests/foreman/ui/test_role.py (hunk 11 above)
Comment (oshtaier) [inline tests/foreman/ui/test_role.py]: nevermind )

**Comment by oshtaier** [comment]
Sorry man for delays from my side. It is just too many of PRs at that moment

**Comment by jyejare** [comment]
@oshtaier , I updated all the tier2 and tier3 tests with case levels.

File: tests/foreman/ui/test_role.py [hunk 14]
```diff
@@ -660,138 +658,827 @@ def test_positive_disable_all_filters_overriding_option(self):
...
+            1. Org Admin should be able to create new users
+            2. Only Org Admin role should be available to assign to its users
+            3. Org Admin should be able to assign Org Admin role to its users
+
+        :CaseLevel: Integration
+        """
+
+    @stubbed
+    @tier2
+    def test_positice_access_users_inside_org_admin_taxonomies(self):
```
Comment (sghai) [inline tests/foreman/ui/test_role.py]: positive ?

**Comment by sghai** [comment]
Looks great w/ so many test scenarios. Please fix the typo and PR is good.

**Comment by sghai** [comment]
ACK

**Comment by oshtaier** [comment]
Good, if that PR accumulated all comments from other Canned Role PRs then good to go
ACK pending @sghai comment

**Comment by jyejare** [comment]
@sghai I fixed your comment, please merge.

**Pull request merged**